libs/MeshKernel/src/CurvilinearGrid/CurvilinearGridCreateUniform.cpp

```diff
@@ -256,25 +225,16 @@ namespace meshkernel
                                                           std::shared_ptr<Polygons> polygons,
                                                           UInt polygonIndex) const
     {
-        if (blockSizeX <= 0.0)
-        {
-            throw AlgorithmError("BlockSizeX cannot be <= 0");
-        }
-        if (blockSizeY <= 0.0)
-        {
-            throw AlgorithmError("BlockSizeY cannot be <= 0");
-        }
+
+        range_check::CheckInOpenInterval(angle, {-90.0, 90.0}, "Grid angle");
+        range_check::CheckGreater(blockSizeX, 0.0, "X block size");
```

Comment:
Should the angle be checked that it is in the closed interval?
The code it replaces was for the closed interval (std::abs(angle) > 90.0).
However, the comment in the original does match your code.

I think it should be either half open or a closed interval

Reply:
@BillSenior `if (std::abs(angle) > 90.0)` is equiv to `if (angle < -90.0 || angle  > 90.0)`. So we want what's in between, -90 <= angle <=90, therefore `CheckInClosedInterval` must be used rather than `CheckInOpenInterval`.